Cleanup Keystone contracts

DeividasK opened the pull request:
Changes:
- Remove `require` in favor of custom errors.
- Change returns to be explicit.
- Update the compile script to work with Foundry. For some reason, `solc` did not work with a library reference. Given that we are using Foundry anyways, it made more sense to update the contract compilation script instead of trying to make it work with `solc`.

archseer commented:
> * Inline Utils inside the contract. Copy and paste the split report into a mock contract.

I've kept it separate since the report parsing code should be a library for downstream consumers (being able to split out metadata vs report). But maybe this is more efficient for inlining?

cds95 commented on contracts/src/v0.8/keystone/KeystoneForwarder.sol:
```diff
@@ -4,77 +4,135 @@ pragma solidity ^0.8.19;
-import {Utils} from "./libraries/Utils.sol";
 
-// solhint-disable gas-custom-errors, no-unused-vars
 contract KeystoneForwarder is IForwarder, ConfirmedOwner, TypeAndVersionInterface {
```
What is the KeystoneForwarder?

DeividasK commented on contracts/src/v0.8/keystone/interfaces/IForwarder.sol:
```diff
@@ -1,5 +0,0 @@
-pragma solidity ^0.8.19;
-
-/// @title IForwarder - forwards keystone reports to a target
-interface IForwarder {}
```
Removed it as it is empty. We can add it later if needed.

DeividasK commented:
> > * Inline Utils inside the contract. Copy and paste the split report into a mock contract.
> 
> I've kept it separate since the report parsing code should be a library for downstream consumers (being able to split out metadata vs report). But maybe this is more efficient for inlining?

It is more efficient to inline, but I like having a library for consumers. I renamed it from Utils and added a comment about it.
